test/ecto_diff_test.exs

```diff
@@ -614,6 +614,312 @@ defmodule EctoDiffTest do
              } = diff
     end
 
+    # Has Many through polymorphic Association
+
+    test "no changes with has_many through" do
+      {:ok, pet} =
+        %{
+          name: "Spot",
+          skills: [%{name: "Eating"}, %{name: "Sleeping"}],
+          resources: [
+            %{toys: [%{name: "ball", type: "play", quantity: 1}, %{name: "mouse", type: "play", quantity: 2}]}
+          ]
+        }
+        |> Pet.new()
+        |> Repo.insert()
+
+      assert {:ok, :unchanged} = EctoDiff.diff(pet, pet)
+    end
+
+    test "insert with has_many through" do
+      {:ok, pet} =
+        %{
+          name: "Spot",
+          resources: [
+            %{toys: [%{name: "ball", type: "play", quantity: 1}, %{name: "mouse", type: "play", quantity: 2}]}
+          ]
+        }
+        |> Pet.new()
+        |> Repo.insert()
+
+      pet = Repo.preload(pet, [:resources, :toys])
+
+      id = pet.id
+      [%{id: resource_id}] = pet.resources
+      [ball_id, mouse_id] = Enum.map(pet.toys, & &1.id)
```

Comment:
:warning: Not 100% certain how Ecto handles the preload in this case, but if it makes a db query then the order in which the toys are returned are not guaranteed. The [docs](https://hexdocs.pm/ecto/Ecto.Repo.html#c:preload/3) have some examples of how to preload an association in a guaranteed order.